manifests/config.pp

```diff
@@ -0,0 +1,36 @@
+# @private
+#
+# This class is used to configure the jmxtrans config files
+#
+class jmxtrans::config {
+  include ::jmxtrans
+
+  if $::jmxtrans::manage_service_file {
+    case $::service_provider {
+      'systemd': {
+        file { '/usr/lib/systemd/system/jmxtrans.service':
+          ensure  => file,
+          owner   => 'root',
+          group   => 'root',
+          mode    => '0444',
+          content => epp('jmxtrans/jmxtrans.service.epp', { 'user' => $::jmxtrans::user }),
+        }
+        ~>
+        exec { 'jmxtrans systemctl daemon-reload':
+          command     => 'systemctl daemon-reload',
+          refreshonly => true,
+          path        => $::path,
+        }
+      }
+      default: {
+        file { '/etc/init.d/jmxtrans':
+          ensure  => file,
+          owner   => 'root',
+          group   => 'root',
+          mode    => '0755',
+          content => epp('jmxtrans/jmxtrans.epp'),
```

Comment:
It might be nice to change the name to clarify that this is an init file. (`jmxtrans/init/jmxtrans.epp`? `jmxtrans/jmxtrans.init.epp`?)